src/TesApi.Web/Runner/TaskToNodeTaskConverter.cs

```diff
@@ -0,0 +1,308 @@
+    /// 2 .- Handles local file paths in inputs when Cromwell is using the local filesystem with a blob FUSE driver.
+    /// </summary>
+    public class TaskToNodeTaskConverter
+    {
+        /// <summary>
+        /// Metrics file name
+        /// </summary>
+        public const string MetricsFileName = "metrics.txt";
+        /// <summary>
+        /// Batch task working directory environment variable
+        /// </summary>
+        public const string BatchTaskWorkingDirEnvVar = "%AZ_BATCH_TASK_WORKING_DIR%";
+
+        private readonly string pathParentDirectory = BatchTaskWorkingDirEnvVar;
+        private readonly string containerMountParentDirectory = BatchTaskWorkingDirEnvVar;
+        private readonly IStorageAccessProvider storageAccessProvider;
+
+        private readonly TerraOptions terraOptions;
+        private readonly ILogger<TaskToNodeTaskConverter> logger;
+
+
+        /// <summary>
+        /// Constructor of TaskToNodeTaskConverter
+        /// </summary>
+        /// <param name="terraOptions"></param>
+        /// <param name="storageAccessProvider"></param>
+        /// <param name="logger"></param>
+        public TaskToNodeTaskConverter(IOptions<TerraOptions> terraOptions, IStorageAccessProvider storageAccessProvider, ILogger<TaskToNodeTaskConverter> logger)
+        {
+            ArgumentNullException.ThrowIfNull(terraOptions);
+            ArgumentNullException.ThrowIfNull(storageAccessProvider);
+            ArgumentNullException.ThrowIfNull(logger);
+
+            this.terraOptions = terraOptions.Value;
+            this.logger = logger;
+            this.storageAccessProvider = storageAccessProvider;
+        }
+
+        /// <summary>
+        /// Parameter-less constructor for mocking
+        /// </summary>
+        protected TaskToNodeTaskConverter() { }
+
+        /// <summary>
+        /// Converts TesTask to a new NodeTask
+        /// </summary>
+        /// <param name="task">Node task</param>
+        /// <param name="additionalInputs"></param>
+        /// <param name="containerCleanupOptions"></param>
+        /// <param name="cancellationToken"></param>
+        public virtual async Task<NodeTask> ToNodeTaskAsync(TesTask task,
+            IList<TesInput> additionalInputs, RuntimeContainerCleanupOptions containerCleanupOptions,
+            CancellationToken cancellationToken)
+        {
+            ArgumentNullException.ThrowIfNull(task);
+
+            try
+            {
+                var builder = new NodeTaskBuilder();
+
+                //TODO: Revise this assumption (carried over from the current implementation) and consider Single() if in practice only one executor per task is supported.
+                var executor = task.Executors.First();
+
+                builder.WithId(task.Id)
+                    .WithWorkflowId(task.WorkflowId)
+                    .WithContainerCommands(executor.Command.Select(EscapeBashArgument).ToList())
```

Comment:
`EscapeBashArgument` is required when each part of the `Command` array from the `TesTask` ends up interpreted by `bash`. If we pass the array directly to the Docker API (which we do here), then they should NOT be escaped.